contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java

```diff
@@ -1015,19 +1020,19 @@ int RegQueryValueEx(HKEY hKey, String lpValueName, int lpReserved,
      *         the function fails, the return value is a nonzero error code
      *         defined in Winerror.h.
      */
-    int RegSetValueEx(HKEY hKey, String lpValueName, int Reserved,
-                  int dwType, Pointer lpData, int cbData);
+    int RegSetValueEx(HKEY hKey, NativeWin32String lpValueName, int Reserved,
+                  int dwType, NativeWin32String lpData, int cbData);
 
     /**
-     * See {@link #RegSetValueEx(com.sun.jna.platform.win32.WinReg.HKEY, java.lang.String, int, int, com.sun.jna.Pointer, int) }
+     * See {@link #RegSetValueEx(HKEY, NativeWin32String, int, int, NativeWin32String, int)}
      */
-    int RegSetValueEx(HKEY hKey, String lpValueName, int Reserved,
-                      int dwType, char[] lpData, int cbData);
+    int RegSetValueEx(HKEY hKey, NativeWin32String lpValueName, int Reserved,
+                      int dwType, Pointer lpData, int cbData);
```

Comment:
This `char[]` version isn't necessarily broken assuming you're calling it with the correct data type and you're in UNICODE, but perhaps should be deprecated, as the `Pointer` version would be preferred.